homeassistant/components/slack/notify.py

```diff
@@ -1,10 +1,13 @@
 """Slack platform for notify component."""
+from __future__ import annotations
```

Comment:
Is this necessary, given that Python 3.7 is our minimum version?

Reply:
No it's not :) I seemed to have added this out of habit. If I'm not mistaken, the next version will require Python 3.8 already and once the mypy update #45485 is merged, the future import is necessary to use `list[int]` and `int | None`.

Reply:
**Correction**
It is necessary. I use a forward reference in the return value for `async_get_service` -> `Optional[SlackNotificationService]`.

Reply:
I'm hesitant to add new imports unless we absolutely have to – how about `"`ing that forward reference?

```python
def async_get_service -> "Optional[SlackNotificationService]":
```

Reply:
I would recommend to use the import. `from __future__ import annotations` is basically just a flag anyway to tell the interpreter (and type checker) to not evaluate type/variable annotations at runtime. See: [cpython/__future__.py](https://github.com/python/cpython/blob/master/Lib/__future__.py). I assume it will be added to a lot of typed modules in the next couple weeks, since it allows the use of `int | None` and `list[int]` starting with Python 3.7 (except in Type Aliases).

It's also already used in core util functions: [util/process.py](https://github.com/home-assistant/core/blob/dev/homeassistant/util/process.py).

Lastly , it will be more cumbersome to change all `" ... "` then to remove the one import, once it's the default.

Reply:
Sounds good!